components/tidb_query_common/src/storage/scanner.rs

```diff
@@ -98,13 +101,14 @@ impl<T: Storage> RangesScanner<T> {
             working_range_begin_key: Vec::with_capacity(KEY_BUFFER_CAPACITY),
             working_range_end_key: Vec::with_capacity(KEY_BUFFER_CAPACITY),
             rescheduler: RescheduleChecker::new(),
+            _phantom: PhantomData,
         }
     }
 
     /// Fetches next row.
     // Note: This is not implemented over `Iterator` since it can fail.
     // TODO: Change to use reference to avoid allocation and copy.
-    pub async fn next(&mut self) -> Result<Option<OwnedKvPair>, StorageError> {
+    pub async fn next(&mut self) -> Result<Option<KeyspaceKv<F>>, StorageError> {
```

Comment:
I have some questions about the abstraction here.

Using `KeyspaceKv` for all types of `KvFormat` seem not the best solution. Is it possible to create a new trait for KvPair and make it an associated type of `KvFormat`?